nixos/modules/tasks/filesystems/bcachefs.nix

```diff
@@ -90,22 +88,39 @@ let
     };
   };
 
+  assertions = [
+    {
+      assertion = let
+        kernel = config.boot.kernelPackages.kernel;
+      in (
+        kernel.kernelAtLeast "6.7" || (
+          lib.hasAttr "BCACHEFS_FS" kernel.structuredExtraConfig && (
+            kernel.structuredExtraConfig.BCACHEFS_FS == lib.kernel.module ||
+            kernel.structuredExtraConfig.BCACHEFS_FS == lib.kernel.yes
+          ))
```

Comment:
```suggestion
          lib.elem (kernel.structuredExtraConfig.BCACHEFS_FS or null) [
            lib.kernel.module lib.kernel.yes
          ]
```

Reply:
Didn't see that, give me a minute to change it.

Reply:
Done.